src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -543,14 +621,15 @@ def end_string(self):
         # pylint: disable=no-self-use
         return "omp end parallel"
 
-    def _get_private_list(self):
+    def _get_private_clause(self):
```

Comment:
Does this method assume a certain sharing default? e.g. default(shared) ? In this case, this PR introduced the OMPDefaultClause which can interfere with this and make this logic wrong if it is not the expected value?

Maybe a TODO in the docstrign and check/raise error is enough for now until we can think this better.

Reply:
I'm not sure - the directive speficially adds a `OMPDefaultClause(clause_type=OMPDefaultClause.DefaultClauseTypes.SHARED))` to match the previous behaviour so I don't think any default is going to come up. I can add it though if you still think its needed.

Reply:
But the children can be changed. So:
```
dir = OMPParallelDirective.create(children)
dir.walk(OMPDefaultClause).replace_with(
    OMPDefaultClause(clause_type=OMPDefaultClause.DefaultClauseTypes.PRIVATE)
)
FortranWriter(dir)
```
Could this be wrong? Because it calls lower_to_language_level which calls _get_private_clause which adds a private clause, but in this case this is redundant and are the SHARED the ones that need to be explicitly listed.

Reply:
It would be yeah - I've added a check for this now in `_get_private_clause` which has to be called for codegen to happen which should handle this.